src/els_general_provider.erl

```diff
@@ -0,0 +1,125 @@
+-module(els_general_provider).
```

Comment:
Shouldn't this be called `els_initialize_provider`? The word general doesn't say much about what it does and IMO gives the idea that this module contains what is in the `els_provider`.

Reply:
The spec lists lifecyle messages under the heading "general". `initialize`, `initialized`, `shutdown`, `exit`.

So this matches the spec, but I think the spec is using a bad naming.

Reply:
What Alan said. I also think is not a great name, but that's what those messages are referred to in the specs.